src/runtime/runtime_internal.h

```diff
@@ -177,6 +182,28 @@ __attribute__((always_inline)) T max(const T &a, const T &b) {
     return a > b ? a : b;
 }
 
+template <typename T>
+__attribute__((always_inline)) T min(const T &a, const T &b) {
+    return a < b ? a : b;
+}
+
+template <typename T, typename U>
+__attribute__((always_inline)) T reinterpret(const U &x) {
+    T ret;
+    memcpy(&ret, &x, min(sizeof(T), sizeof(U)));
+    return ret;
+}
+
+// http://www.cse.yorku.ca/~oz/hash.html
+template <typename Iterator>
```

Comment:
Please add a comment about what this hash is expected to be used for, and what the limits of its usefulness will be; otherwise people may add inappropriate uses without realizing it.